libraries/chain/include/eosio/chain/host_context.hpp

```diff
@@ -565,7 +565,13 @@ class host_context {
 
    /// Sync call methods:
 
-   // sync calls can be initiated from actions or other sync calls
+   enum class call_error_code : int64_t {
+      sync_call_not_supported_by_receiver = -1, // receiver contract does not have sync_call entry point
+      unsupported_data_version            = -2, // version in the data header is not supported
+      called_function_not_found           = -3, // called function is not found in the receiver contract
+      empty_receiver                      = -4  // receiver contract code is empty
+   };
+   // Negative status indicates failure, positive or 0 is the size of return value of the called function
```

Comment:
btw I noticed now that maybe these should be `FC_REFLECT_ENUM`ed so they are output pretty in the trace instead of just a number. Assuming we can `FC_REFLECT_ENUM` negatives.

Reply:
I tested. Negative enums worked with FC_REFLECT_ENUM.

https://github.com/AntelopeIO/spring/pull/1525/commits/92175f6147f18d4cd81a8535288003a35ae20545